mycore-viewer/src/main/typescript/modules/base/MyCoReViewer.ts

```diff
@@ -16,113 +16,93 @@
-        constructor(private _container:JQuery, private _settings:MyCoReViewerSettings) {
-            this._eventHandlerMap = new MyCoReMap<string, Array<mycore.viewer.components.ViewerComponent>>();
-            this._components = new Array<mycore.viewer.components.ViewerComponent>();
-            this._initializingEvents = new Array<mycore.viewer.widgets.events.ViewerEvent>();
-
-            this.initialize();
-        }
 
+import {MyCoReMap, VIEWER_COMPONENTS, ViewerFormatString} from "./Utils";
+import {MyCoReViewerSettings} from "./MyCoReViewerSettings";
+import {ViewerComponent} from "./components/ViewerComponent";
+import {ViewerEvent} from "./widgets/events/ViewerEvent";
+import {WaitForEvent} from "./components/events/WaitForEvent";
 
-        private _eventHandlerMap:MyCoReMap<string, Array<mycore.viewer.components.ViewerComponent>>;
-        private _initializing:boolean = false;
-        private _initializingEvents:Array<mycore.viewer.widgets.events.ViewerEvent>;
-        private _components:Array<mycore.viewer.components.ViewerComponent>;
 
-        public get container() {
-            return this._container;
-        }
+export class MyCoReViewer {
 
-        public addComponent(ic:mycore.viewer.components.ViewerComponent) {
-            var that = this;
-            ic.bind(function (event:mycore.viewer.widgets.events.ViewerEvent) {
-                that.eventTriggered(event);
-            });
+    constructor(private _container: JQuery, private _settings: MyCoReViewerSettings) {
+        this._eventHandlerMap = new MyCoReMap<string, Array<ViewerComponent>>();
+        this._components = new Array<ViewerComponent>();
+        this._initializingEvents = new Array<ViewerEvent>();
 
-            var events = ic.handlesEvents;
+        this.initialize();
+    }
 
-            if (typeof events != "undefined" && events != null && events instanceof Array) {
-                events.push(components.events.WaitForEvent.TYPE)
 
-                for (var eIndex in events) {
-                    var e = events[ eIndex ];
-                    // initialize the array for the event first
-                    if (!this._eventHandlerMap.has(e)) {
-                        this._eventHandlerMap.set(e, new Array<mycore.viewer.components.ViewerComponent>());
-                    }
+    private _eventHandlerMap: MyCoReMap<string, Array<ViewerComponent>>;
+    private _initializing: boolean = false;
+    private _initializingEvents: Array<ViewerEvent>;
+    private _components: Array<ViewerComponent>;
 
-                    this._eventHandlerMap.get(e).push(ic);
-                }
-            } else {
-                console.log(ViewerFormatString("The component {comp} doesnt have a valid handlesEvents!", {comp : ic}))
-            }
+    public get container() {
+        return this._container;
+    }
 
-            this._components.push(ic);
-            ic.init();
-        }
+    public addComponent(ic: ViewerComponent) {
+        var that = this;
```

Comment:
Use let instead of var.
You also don't need "that" if you use () => syntax
```
ic.bind(event: ViewerEvent => {
            this.eventTriggered(event);
        });
```